src/pages/index.tsx

```diff
@@ -1,18 +1,18 @@
 import Image from "next/image";
 import { Inter } from "next/font/google";
-import Navbar from "./navabar";
+import Navbar from "../components/navabar";
 import { useState } from "react";
+import Link from "next/link";
+import MainSection from "../components/main_section";
 export type ColorMode = "Dark Mode" | "Light Mode";
```

Comment:
Separation of concern suggests that you should organize your code in a way that clearly separates different aspects of your application.

One common approach is to have a dedicated file or module where you declare all your **interfaces**, **enums**, and **custom data types**. This helps keep your codebase organized and makes it easier to find and reference these shared definitions.

Alternatively, if you prefer to have these type definitions inline with the code that uses them, it's important to use clear visual separation, such as blank lines, to demarcate where the type definitions end and the rest of the implementation begins.